### PR TITLE
feat: add ID's to `<style>` tags

### LINKED ISSUE #216: Perfect the custom attributes, Built-in component information, such as: name, version.
## why?

I have many components.

 
I would like to distinguish between which component and the version number to use.

I found the `attrs`

```javascript
require('style-loader?{attrs:{id: "style-tag-id"}}!style.css');
```

But the information of the component can only be obtained in the compilation of each file.

so

I would like to make this change in 'style-loader/index.js'.

```javascript
	var findPackage = require('find-package');
        ...
	if(this.cacheable) this.cacheable();
	var query = loaderUtils.getOptions(this) || {};
	var pkg = findPackage(this.resourcePath) || {};
	if (query.attrs) {
		Object.keys(query.attrs).forEach(function(key) {
			query.attrs[key] = query.attrs[key].replace(/\[name\]/ig, pkg.name).replace(/\[version\]/ig, pkg.version);
		});
	}
        return ...
```

then

I use it like this

```
require('style-loader?{attrs:{component: "[name]@[version]"}}!style.css');
```

The result:

If it is acceptable ?

### PR DESCRIPTION
**Pull request by alex-mm**
fix: #216

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -14,6 +14,15 @@ module.exports.pitch = function (request) {
 
 	var options = loaderUtils.getOptions(this) || {};
 
+	var pkg = findPackage(this.resourcePath) || {};
+  if (options.attrs) {
+    Object.keys(options.attrs).forEach(function(key) {
+			options.attrs[key] = options.attrs[key]
+				.replace(/\[name\]/ig, pkg.name)
+				.replace(/\[version\]/ig, pkg.version);
+    });
```
[the remark this answers is not in the document]
Comment (alex-mm) [reply index.js]: 1. `find-package`  has been removed.
2. But loader-utils.interpolateName is unsatisfactory. We need to get the version number of the package corresponding to the current file.
3.  is there a better way to get it than from package.json?

**Comment by evilebottnawi** [comment]
Why you need version here? Better add this attribute using own configuration, it is not universal solution

**Comment by michael-ciniawsky** [comment]
Closing due to inactivity feel free to repoen once the requested chances are addressed :)

**Comment by m-orsh** [comment]
Lets say I have 2 standalone apps. Each one needs to check whether the css libraries it needs have been loaded prior to loading them. If I load app 1, and this loads some css libs that app 2 also will use, without a method of identifying the css I have imported, how can I identify if the css has already been loaded? I think the only good way is with this feature.

**Pull request closed without merging**